EIPS/eip-7867.md

```diff
@@ -0,0 +1,553 @@
+---
+eip: 7867
+title: Flow Control Wallet Call Capability
+description: An EIP-5792 capability providing atomicity and flow control configuration.
+author: Sam Wilson (@SamWilsn) <sam@binarycake.ca>
+discussions-to: https://ethereum-magicians.org/t/wallet-sendcalls-capability-flow-control/22624
+status: Draft
+type: Standards Track
+category: Interface
+created: 2025-01-17
+requires: 5792
+---
+
+## Abstract
+
+This proposal extends [EIP-5792](./eip-5792.md) to allow dapps to downgrade their required atomicity guarantees and control the behaviour after a failed/reverted call. It introduces the batch-scope concept of `strict` vs. `loose` atomicity, where a `strict` batch remains atomic in the face of chain reorgs and a `loose` batch does not; and the per-call ability to continue after a failed/reverted call (`continue`) or stop processing (`halt`).
+
+## Motivation
+
+While the base EIP-5792 specification works extremely well for smart contract wallets, it does not allow the expression of the full range of flow control options that wallets can implement. For example, a dapp may only be submitting a batch for gas savings and not care about whether all calls are reverted on failure. A wallet may only be able to offer a limited form of atomicity through block builder backchannels, but that may be sufficient for a trading platform.
```

Comment:
> a dapp may only be submitting a batch for gas savings and not care about whether all calls are reverted on failure
Why wouldn't dapp care? What's the user flow in this case?

Reply:
Something like a token dispersement or several unrelated swaps. For example, I could be swapping several ERC-20 tokens into ETH. I just want as much of that ETH as possible, and I'll resubmit any failed swaps.